types/testingutils/qbft_msgs.go

```diff
@@ -188,6 +188,23 @@ var TestingPrepareMessageWithFullData = func(sk *bls.SecretKey, id types.Operato
 	ret.FullData = fullData
 	return ret
 }
+var TestingPrepareMessageWithRoundAndFullData = func(
+	sk *bls.SecretKey,
+	id types.OperatorID,
+	round qbft.Round,
+	fullData []byte,
+) *qbft.SignedMessage {
+	msg := &qbft.Message{
+		MsgType:    qbft.PrepareMsgType,
+		Height:     qbft.FirstHeight,
+		Round:      round,
+		Identifier: TestingIdentifier,
+		Root:       sha256.Sum256(fullData),
+	}
+	ret := SignQBFTMsg(sk, id, msg)
+	ret.FullData = fullData
+	return ret
+}
```

Comment:
This might not be in the scope of this PR... But maybe we should change `TestingPrepareMessageWithParams` to accept `fullData`?

Reply:
I agree that this needs to be done, but that's actually not the only thing that needs to be done among message generators and it will cause a huge amount of conflicts. This PR is open for a while so I'm afraid this may introduce huge merge conflicts. I'd prefer doing this change in a separate PR and merge in quickly (it should be easy to review such a PR and make sure it doesn't break anything because all tests will pass)